src/main/java/io/vavr/collection/Traversable.java

```diff
@@ -1704,6 +1704,45 @@ default Number sum() {
      */
     <U> Traversable<U> zipWithIndex(BiFunction<? super T, ? super Integer, ? extends U> mapper);
 
+    /**
+     * Converts this collection to a {@link Map}.
+     *
+     * @param f A function that maps an element to a key/value pair represented by Tuple2
+     * @param merge A function that merges values that are associated with the same key
+     * @param <K>         The key type
+     * @param <V>         The value type
+     * @return A new {@link HashMap}.
+     */
+    default <K, V, E extends Function<? super T, ? extends Tuple2<? extends K, ? extends V>>> Map<K, V> toMap(
+            E f,
+            BiFunction<? super V, ? super V, ? extends V> merge) {
+        Objects.requireNonNull(f, "f is null");
+        Objects.requireNonNull(merge, "merge is null");
+        return this
+                .map(f)
+                .<K>groupBy(Tuple2::_1)
+                .mapValues(s -> s.<V>map(Tuple2::_2).reduce(merge));
+    }
+
+    /**
+     * Converts this collection to a {@link Map}.
+     *
+     * @param keyMapper   A function that maps an element to a key
+     * @param valueMapper A function that maps an element to a value
+     * @param merge A function that merges values that are associated with the same key
+     * @param <K>         The key type
+     * @param <V>         The value type
+     * @return A new {@link HashMap}.
+     */
+    default <K, V> Map<K, V> toMap(
+            Function<? super T, ? extends K> keyMapper,
+            Function<? super T, ? extends V> valueMapper,
+            BiFunction<? super V, ? super V, ? extends V> merge) {
+        Objects.requireNonNull(keyMapper, "keyMapper is null");
+        Objects.requireNonNull(valueMapper, "valueMapper is null");
+        Objects.requireNonNull(merge, "merge is null");
+        return toMap(t -> Tuple.of(keyMapper.apply(t), valueMapper.apply(t)), merge);
```

Comment:
```java
return map(f)
        .groupBy(keyMapper)
        .mapValues(s -> s.map(valueMapper).reduce(merge));
```

Reply:
Done, but I still need the type witnesses to get this to compile.